week_4/assignment/voting-api/src/app.service.ts

```diff
@@ -48,29 +40,19 @@ export class AppService {
           proposal.voteCount
         } Votes on proposal: ${ethers.utils.parseBytes32String(proposal.name)}`
       );
-      proposals.push(ethers.utils.parseBytes32String(proposal.name));
-      await sleep(500); // to avoid hitting api limits
+      proposals.push({
+       name: ethers.utils.parseBytes32String(proposal.name),
+       votes: `${ethers.utils.formatUnits(proposal.voteCount.toString())} ETH`
+      });
     }
   
-    // Finds and Logs the winner
-    //console.log("waiting before checking winner");
-    //await sleep(1000); // to avoid hitting api limits
-    //console.log("checking winner");
     const winner = await this.ballotContract.winnerName();
-  
-  
-    //const targetBlockNumber = await this.ballotContract.targetBlockNumber();
-    //console.log({ targetBlockNumber });
-  
-    //const winnerName = ethers.utils.parseBytes32String(winner);
-    //console.log(`The winner is: ${winnerName}`);
-
-    //result.push(proposals)
+    const winnerName = ethers.utils.parseBytes32String(winner);
 
-    return proposals
+    result.proposals = proposals;
+    result.winner = winnerName;
+;
+    return result;
```

Comment:
Now the api should return the following object.

<img width="319" alt="Screenshot 2023-03-12 at 15 13 57" src="https://user-images.githubusercontent.com/42499/224554035-e95c94f9-15c1-4bdf-af36-8569a546900d.png">